server/data/alertsApiClient.ts

```diff
@@ -18,6 +18,8 @@ export default class AlertsApiRestClient implements AlertsApiClient {
       size: queryParams?.showAll ? 9999 : 20,
       ...queryParams,
     }
+    delete params.showAll
+
```

Comment:
To avoid mutating state, would something like this work?

```
    const { showAll, ...rest } = queryParams
    const params: AlertsApiQueryParams = {
      size: showAll ? 9999 : 20,
      ...rest,
    }
    ```